modules/mapper-extras/src/main/java/org/opensearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -414,6 +424,11 @@ public ParametrizedFieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName(), ignoreMalformedByDefault, coerceByDefault).init(this);
     }
 
+    @Override
+    public boolean isMultivalue() {
+        return multivalued.explicit() && multivalued.value() != null && multivalued.value();
+    }
```

Comment:
I'm a little concerned with the idea of treating "unset" the same as `false`. 

In particular, existing fields may or may not be multi-valued. We just don't know. I think we should/must treat the `null` (or unset) case as ambiguous.

In my opinion, the more interesting case is when a user defines a new mapping and explicitly says `"multivalued":false`. At that point, if the field is specified multiple times in a document, or specified as an array, we should throw an exception. If a field is explicitly specified as `"multivalued":true`, I'm inclined to accept a single value, not passed in an array -- but I would always return it wrapped in an array.

@normanj-bitquill -- what do you think? I know it's different from the approach you took, but I think it will provide both backward-compatibility and let us clearly identify single-valued fields in future.

Reply:
> In my opinion, the more interesting case is when a user defines a new mapping and explicitly says `"multivalued":false`. At that point, if the field is specified multiple times in a document, or specified as an array, we should throw an exception. 

I am inclined to agree, except I suggest that we make the system more forgiving.  If `"multivalued":false`, then if the field is in fact multi-valued, then we should return the first item encountered without guaranteeing any order.  This may mean that the user will get back different values from different requests, but that's what they're explicitly requesting (any value is good enough).  

> If a field is explicitly specified as `"multivalued":true`, I'm inclined to accept a single value, not passed in an array -- but I would always return it wrapped in an array.

This seems more obvious and straight-forward.

Reply:
@msfroh I'm OK with inverting the ingestion check. By that I mean only give an error when a field contains multiple values, but `"multivalued": false` is set. No errors in other cases. @acarbonetto was curious about if we could do anything different with the results that are returned. Once that discussion is resolved, I'll update this PR.

Reply:
Closing this PR as discussed. If there ever is a fix related to this in the future it would be around how aggregates handle multi-valued fields.

Reply:
> For example, if a min/max aggregation is performed on a field that has multivalued=true, I would expect that the results would be an array/multivalued.

I think that makes a lot of sense. Part of the original motivation for the #16420 was that @anirudha was asking how the SQL plugin could return results in the appropriate format.

Moving forward, I would love to change the default behavior from "unknown" to `multivalued=false`, so that you would need to explicitly mark a multivalued field as such. Unfortunately, that's a (potentially) disruptive change for dynamically-added fields. I'll write down some more thoughts on #16420 on how to handle that.

Reply:
@normanj-bitquill -- do you plan to open another PR to flip the logic?

Once we have the property in place, we should be able to use it for rendering at query time (especially from the SQL plugin).

Incidentally, I think I would add `multivalued` as a `Boolean` field on `MappedFieldType`, since I believe it could be applicable to every subclass.

Reply:
@YANG-DB suggested `mapping._meta#field.name.type:array`.  Is that better than `multivalued`?  
https://github.com/opensearch-project/sql/issues/3138#issuecomment-2546516082

Reply:
Oh, that should at least address the SQL side of things.

I would still like to address this in OpenSearch core, as we move towards a more strictly-defined schema. (There are some indexing ideas that I've been discussing with @backslasht that involve packing fixed-width values for fields, which means that you *need* to know if a field is single- or multi-valued.) That said, if we have a workaround for the more immediate SQL problem, that's a lower priority.

Thanks, @acarbonetto and @normanj-bitquill!